File: app/src/main/java/com/example/mobile6/ui/medicine/MedicationWarningDialogFragment.java
package com.example.mobile6.ui.medicine;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;

import com.example.mobile6.databinding.DialogMedicationWarningBinding;
import com.example.mobile6.ui.base.BaseDialog;

public class MedicationWarningDialogFragment extends BaseDialog<DialogMedicationWarningBinding> {

    public static final String REQUEST_KEY = "medication_warning_request_key";
    public static final String RESULT_KEY = "medication_warning_result";
    public static final String RESULT_CONTINUE = "continue";
    public static final String RESULT_REVIEW = "review";

    @Override
    protected DialogMedicationWarningBinding inflateBinding(@NonNull LayoutInflater layoutInflater, ViewGroup container) {
        return DialogMedicationWarningBinding.inflate(layoutInflater, container, false);
    }

    @Override
    protected void initViews() {
        super.initViews();

        // Set continue button click listener
        binding.btnWarningContinue.setOnClickListener(v -> {
            Bundle result = new Bundle();
            result.putString(RESULT_KEY, RESULT_CONTINUE);
            dismiss(REQUEST_KEY, result);
        });

        // Set review button click listener
        binding.btnWarningReview.setOnClickListener(v -> {
            Bundle result = new Bundle();
            result.putString(RESULT_KEY, RESULT_REVIEW);
            dismiss(REQUEST_KEY, result);
        });
    }
} 

File: app/src/main/java/com/example/mobile6/ui/medicine/MedicationWarningDialogFragment.kt
package com.example.mobile6.ui.medicine

import android.os.Bundle
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import com.example.mobile6.databinding.FragmentMedicationWarningDialogBinding
import com.example.mobile6.ui.base.BaseDialog
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MedicationWarningDialogFragment : BaseDialog<FragmentMedicationWarningDialogBinding>() {

    private var medicineId: Long = 0L
    private var message: String = ""
    private val viewModel: MedicineDetailViewModel by viewModels()

    override val bindingInflater: (LayoutInflater, ViewGroup?) -> FragmentMedicationWarningDialogBinding
        get() = { inflater, container ->
            FragmentMedicationWarningDialogBinding.inflate(inflater, container, false)
        }

    override fun processArguments(args: Bundle) {
        medicineId = arguments?.getLong("medicineId") ?: 0L
        message = arguments?.getString("interactionMessage") ?: ""
    }

    override fun initViews() {
        binding.tvWarningMessage.text = message

        binding.btnWarningReview.setOnClickListener {
            dismiss()
        }

        binding.btnWarningContinue.setOnClickListener {
            val bundle = Bundle().apply {
                putSerializable("selectedMedicine", viewModel.uiState.value.medicine)
            }
            parentFragmentManager.setFragmentResult("prescriptionDataSelectMedicine", bundle)
            dismiss()
            findNavController().navigateUp()
            findNavController().navigateUp()
        }
    }

    override fun initObservers() {
    }
}